Bumped bufr version to 12.0.1

srherbener opened the pull request:
### Summary

This PR bumps the bufr version in common package.py to version 12.0.1. 

### Testing

I tested building on my Mac and was able to successfully build bufr@12.0.1.

### Applications affected

This affects JEDI ioda converters, and other applications that use the NCEPLIBS-bufr package.

### Systems affected

List all systems intentionally or unintentionally affected by this PR.

### Dependencies

- [x] waiting on jcsda/spack/pull/330

### Issue(s) addressed

This partially addresses #778

### Checklist
- [x] This PR addresses one issue/problem/enhancement, or has a very good reason for not doing so.
- [ ] These changes have been tested on the affected systems and applications.
- [ ] All dependency PRs/issues have been resolved and this PR can be merged.

srherbener committed:
Bumped bufr version to 12.0.1, change .gitconfig urls for testing

climbfuji reviewed (dismissed):
bufr@12.0.1 is not in the release/1.5.1 spack submodule, and also not in jcsda_emc_spack_stack

climbfuji commented:
> bufr@12.0.1 is not in the release/1.5.1 spack submodule, and also not in jcsda_emc_spack_stack

Dang, my mistake, just saw your spack PR. I'll dismiss my review and look more carefully again. CI is failing, though.

srherbener committed:
Merge branch 'develop' into feature/bufr-12.0.1

srherbener commented:
I'm getting these error messages for the ubuntu-ci-x86_64-build test:
```
==> Installing bufr-12.0.1-n2tv7iwsukza4qaehstd3r3keo6dspx5
==> No binary for bufr-12.0.1-n2tv7iwsukza4qaehstd3r3keo6dspx5 found: installing from source
Reversed (or previously applied) patch detected!  Assume -R? [n] 
Apply anyway? [n] 
1 out of 1 hunk ignored -- saving rejects to file python/CMakeLists.txt.rej
==> Fetching file:///home/ubuntu/spack-stack/source-cache/_source-cache/archive/52/525f26238dba6511a453fc71cecc05f59e4800a603de2abbbbfb8cbb5adf5708.tar.gz
==> Patch /home/ubuntu/spack-stack/CI/actions-runner/_work/spack-stack/spack-stack/spack/var/spack/repos/builtin/packages/bufr/python-version.patch failed.
==> Error: ProcessError: Command exited with status 1:
    '/usr/bin/patch' '-s' '-p' '1' '-i' '/home/ubuntu/spack-stack/CI/actions-runner/_work/spack-stack/spack-stack/spack/var/spack/repos/builtin/packages/bufr/python-version.patch' '-d' '.'
```
I'm not sure what the problem is. I ran into this with manually building on my Mac, but in that case the prompt I got was:
```
Reversed (or previously applied) patch detected!  Assume -R? [y]'
```
(ie, the default response is yes instead of no) and by responding with a `return` the build took off and completed successfully.

I must be missing something in my `feature/burf-12.0.1` branches, but I'm not sure what to change. Any ideas?

Thanks!

climbfuji commented:
Looks like one of the patches that spack applies can't be applied cleanly. Maybe it's been fixed in the source code.

srherbener committed:
Updated spack submodule commit hash

srherbener commented:
> Looks like one of the patches that spack applies can't be applied cleanly. Maybe it's been fixed in the source code.

Thanks @climbfuji! I think I've got it fixed. The python patch is no longer needed for bufr@12.0.1, so I updated the bufr package.py accordingly.

climbfuji commented:
@srherbener I merged https://github.com/JCSDA/spack/pull/330, please update your github submodule pointer and also updated this branch from develop. Thanks!

srherbener committed:
Merge remote-tracking branch 'jcsda/develop' into feature/bufr-12.0.1

srherbener committed:
Updated spack submodule hash

srherbener committed:
Reverted .gitmodules config to point to the jcsda_emc_spack_stack base branch in our spack
repo fork.

srherbener commented:
> @srherbener I merged [JCSDA/spack#330](https://github.com/JCSDA/spack/pull/330), please update your github submodule pointer and also updated this branch from develop. Thanks!

Done! I also reverted the .gitmodules back to point to the jcsda_emc_spack_stack branch in our spack fork.
